### ISSUE TITLE
hoeffding_tree_test.cpp keep warning: moving a temporary object prevents copy elision

### ISSUE THREAD

**Issue by robotcator** [mlpack/mlpack#1911]
#### Issue description

Warning in hoeffding_tree_test.cpp when compiling the mlpack. 

```
mlpack/src/mlpack/tests/main_tests/hoeffding_tree_test.cpp:512:29: warning: 
      moving a temporary object prevents copy elision [-Wpessimizing-move]
  SetInputParam("training", std::move(std::make_tuple(info, inputData)));
                            ^
/Users/robotcator/clone/boost/dev-mlpack/mlpack/src/mlpack/tests/main_tests/hoeffding_tree_test.cpp:512:29: note: 
      remove std::move call here
  SetInputParam("training", std::move(std::make_tuple(info, inputData)));
```

#### Your environment

 * version of mlpack: mlpack master branch 
 * operating system: MacOS Sierra 10.12.6
 * compiler: clang 
 * version of dependencies (Boost/Armadillo): boost 1.68

**Comment by rcurtin** [comment]
I think it's just fine to remove the `std::move()` there like the warning suggests---want to open a PR with that change?  If not I can handle it.

**Comment by robotcator** [comment]
> I think it's just fine to remove the `std::move()` there like the warning suggests---want to open a PR with that change? If not I can handle it.

Okay, done in [PR1914](https://github.com/mlpack/mlpack/pull/1914).

**Comment by jeffin143** [comment]
@robotcator , since it is done , we can close the issue :). Thanks for the issue and fixes

**Comment by rcurtin** [comment]
Fixed in #1914. Thanks! :+1:
